sources/HUBViewModelDiff.m

```diff
@@ -57,7 +64,9 @@ + (instancetype)diffFromViewModel:(id<HUBViewModel>)fromViewModel
                         algorithm:(HUBDiffAlgorithm)algorithm
 {
     NSParameterAssert(algorithm);
-    return algorithm(fromViewModel, toViewModel);
+    HUBViewModelDiff *diff =  algorithm(fromViewModel, toViewModel);
+    [diff calculateHeaderChangesFromViewModel:fromViewModel toViewModel:toViewModel];
+    return diff;
```

Comment:
Isn't the header comparison totally separate from the diffing algorithm though? It's only possible to assign a single header component so why would we want every algorithm to make the same comparison? Maybe I've misunderstood what the diffing algorithm is doing...